### PR TITLE
Testing part 2

### PR DESCRIPTION
**Pull request by muffintoxin**

### CODE REVIEW

File: test/fancy_select/index.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+const { createElement } = require('react');
+const { assert } = require('chai');
+const { mount, shallow } = require('enzyme');
+
+const FancySelect = require('../../lib/fancy_select').default;
+
+class Child extends FancySelect {
```
Comment (atomiomi) [inline test/fancy_select/index.js]: no need in Child, FancySelect is not a base class, it implements render method

File: test/fancy_select/index.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      ref: this.setEl('input'),
+      type: 'text',
+      value: this.state.value,
+    });
+  }
+}
+
+
+describe('<FancySelect />', () => {
+  it('renders FancySelect', () => {
```
Comment (atomiomi) [inline test/fancy_select/index.js]: put this on one `describe`

File: test/fancy_select/index.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    const props = {
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    const wrapper = mount(createElement(FancySelect, props));
+    assert.equal(wrapper.find('ul .c-fancy_select-list').length, 1, 'FancySelect was rendered');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: also check that 2 `c-fancy_select-item` were rendered

File: test/fancy_select/index.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  it('defaultValue', () => {
+    const props = {
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 555 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    wrapper = shallow(createElement(Child, props));
+    assert.equal(wrapper.instance().state.value, 555, 'default value is 555');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: check `index` field

File: test/fancy_select/index.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    wrapper = shallow(createElement(Child, props));
+
+    wrapper.instance().setValue('value');
+    assert.equal(wrapper.instance().state.value, 'value', 'set value');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: check `index` field from state

File: test/fancy_select/index.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    wrapper = shallow(createElement(Child, props));
+    const instance = wrapper.instance();
+    instance.setValue('value');
+
+    assert.equal(instance.getValue(), 'value', 'value was updated');
+    assert.isFalse(instance.isPristine(), 'not pristine');
+
+    instance.setPristine();
+
+    assert.equal(instance.getValue(), 'value', 'value is same');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: also  check that `index` wasn't changed

File: test/fancy_select/index.js (hunk 3 above)
Comment (atomiomi) [reply test/fancy_select/index.js]: check that clicking on `c-fancy_select-item` changes value

Comment (atomiomi) [reply test/fancy_select/index.js]: check that `c-fancy_select-item` with active value has `is-active` class

File: test/fancy_select/index.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+      ],
+    };
+    const wrapper = mount(createElement(FancySelect, props));
+    assert.equal(wrapper.find('ul .c-fancy_select-list').length, 1, 'FancySelect was rendered');
+    assert.equal(wrapper.find('li .c-fancy_select-item').length, 2, 'two options were rendered');
+    assert.equal(wrapper.instance().state.index, 0, 'index is 0');
+
+    wrapper.find('li .c-fancy_select-item').last().simulate('click');
+
+    assert.equal(wrapper.instance().state.value, 2, 'value was changed');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: Keep instance in variable

File: test/fancy_select/index.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
+const { createElement } = require('react');
+const { assert } = require('chai');
+const { mount, shallow } = require('enzyme');
+
+const FancySelect = require('../../lib/fancy_select').default;
+
+
+describe('<FancySelect />', () => {
```
Comment (atomiomi) [inline test/fancy_select/index.js]: One describe for module

File: test/fancy_select/index.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    const props = {
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    const wrapper = shallow(createElement(FancySelect, props));
+    const instance = wrapper.instance();
+    instance.setValue('value');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: Why did u set value that is not in options?

File: test/fancy_select/index.js [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+    assert.equal(instance.getValue(), 'value', 'value was updated');
+    assert.equal(instance.state.index, -1, 'index is -1');
+    assert.isFalse(instance.isPristine(), 'not pristine');
+
+    instance.setPristine();
+
+    assert.equal(instance.getValue(), 'value', 'value is same');
+    assert.equal(instance.state.index, -1, 'index is the same');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: Same here

File: test/form_group/index.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    const wrapper = mount(createElement(FormGroup));
+    assert.isTrue(wrapper.find('div').hasClass('c-form_group'));
+
+    wrapper.unmount();
+  });
+
+  it('should render the children', () => {
+    const props = { children: ['one', 'two', 'three'] };
+    const wrapper = mount(createElement(FormGroup, props));
+    assert.equal(wrapper.find('div .c-form_group-item').length, 3, 'children were rendered');
```
Comment (atomiomi) [inline test/form_group/index.js]: Check that component has corresponding class

File: test/radio/index.js [hunk 12]
```diff
@@ -0,0 +1,52 @@
+const { createElement } = require('react');
+const { assert } = require('chai');
+const { mount, shallow } = require('enzyme');
+const { fake } = require('sinon');
+
+const Radio = require('../../lib/radio').default;
+
+
+describe('<Radio />', () => {
```
Comment (atomiomi) [inline test/radio/index.js]: One describe per module

File: test/fancy_select/index.js (hunk 6 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: +

File: test/fancy_select/index.js (hunk 7 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: +

File: test/fancy_select/index.js (hunk 9 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: `+`

File: test/fancy_select/index.js (hunk 10 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: `+`

File: test/radio/index.js (hunk 12 above)
Comment (muffintoxin) [reply test/radio/index.js]: `+`

File: test/form_group/index.js (hunk 11 above)
Comment (muffintoxin) [reply test/form_group/index.js]: `+`

File: test/fancy_select/index.js [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+    assert.equal(wrapper.find('li .c-fancy_select-item').length, 2, 'two options were rendered');
+    assert.equal(instance.state.value, 1, 'value is set');
+    assert.equal(instance.state.index, 0, 'index is set');
+
+    wrapper.find('li .c-fancy_select-item').last().simulate('click');
+
+    assert.equal(instance.state.value, 2, 'value was changed');
+    assert.equal(instance.state.index, 1, 'index was changed');
+    assert.isTrue(
+      wrapper.find('li .c-fancy_select-item').last().hasClass('is-active'),
```
Comment (atomiomi) [inline test/fancy_select/index.js]: `move wrapper.find('li .c-fancy_select-item').last()` to variable;

File: test/fancy_select/index.js [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+      'active option has active class',
+    );
+
+    wrapper.unmount();
+  });
+
+  it('defaultValue', () => {
+    const props = {
+      defaultValue: 1,
```
Comment (atomiomi) [inline test/fancy_select/index.js]: U need to check 2 cases:
- when defaultValue is passed
- when there is no defaultValue or it's not in options.

File: test/fancy_select/index.js [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 222 },
+      ],
+    };
+    const wrapper = shallow(createElement(FancySelect, props));
+    const instance = wrapper.instance();
+
+    assert.equal(instance.getValue(), 1, 'default value is set');
+    assert.equal(instance.state.index, 0, 'index is correct');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: why do u check defaultValue here? u checked it in case above

File: test/fancy_select/index.js [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+      defaultValue: 1,
+      options: [
+        { imageClass: 'c-sandbox-fancy_image_1', key: 'Eins', value: 1 },
+        { imageClass: 'c-sandbox-fancy_image_2', key: 'Zwei', value: 2 },
+      ],
+    };
+    const wrapper = shallow(createElement(FancySelect, props));
+    const instance = wrapper.instance();
+
+    assert.equal(instance.getValue(), 1, 'default value is set');
```
Comment (atomiomi) [inline test/fancy_select/index.js]: again no need to test defaultValue

File: test/form_group/index.js [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+  it('should render the children', () => {
+    const props = { children: ['one', 'two', 'three'] };
+    const wrapper = mount(createElement(FormGroup, props));
+
+    assert.equal(wrapper.find('div .c-form_group-item').length, 3, 'all children were rendered');
+    assert.isTrue(
+      wrapper
+        .find('div .c-form_group')
+        .hasClass(`is-multiple is-size-${props.children.length}`,
+        ));
```
Comment (atomiomi) [inline test/form_group/index.js]: what with formatting?

File: test/fancy_select/index.js (hunk 13 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: `+`

File: test/form_group/index.js (hunk 17 above)
Comment (muffintoxin) [reply test/form_group/index.js]: `+`

File: test/fancy_select/index.js (hunk 16 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: `+`

File: test/fancy_select/index.js (hunk 15 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: `+`

File: test/fancy_select/index.js (hunk 13 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: doesn't work hier

File: test/fancy_select/index.js (hunk 14 above)
Comment (muffintoxin) [reply test/fancy_select/index.js]: `+`

File: test/fancy_select/index.js [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+      'active option has active class',
+    );
+
+    wrapper.unmount();
+  });
+
+  it('defaultValue', () => {
+    // default value is set
+    const props = {
+      defaultValue: 1,
```
Comment (atomiomi) [inline test/fancy_select/index.js]: default value is also not from options

**Pull request merged**